legate/core/__init__.py

```diff
@@ -123,3 +123,25 @@
     ReductionOp,
 )
 from .io import CustomSplit, TiledSplit, ingest
+
+import numpy as _np
+import random as _random
+from typing import Any
+from .runtime import AnyCallable
+
+_np.random.seed(1234)
+_random.seed(1234)
+
+
+def _warn_seed(func: AnyCallable) -> AnyCallable:
+    def wrapper(*args: Any, **kw: Any) -> Any:
+        print(
```

Comment:
Can we use `warnings.warn` instead, so this message is not reported multiple times?

Reply:
I was thinking actually to use the python `warnings` module https://docs.python.org/3/library/warnings.html#warnings.warn, which makes sure to deduplicate messages. AFAICT what you're doing now just introduces formatting, and doesn't take advantage of the rest of the functionality of the `warnings` module.

Reply:
done